tests/functional/object/rgw/test_obc_quota.py

```diff
@@ -117,3 +121,80 @@ def test_obc_quota(
                 logger.error("Copy objects to bucket failed unexpectedly!!")
         else:
             logger.info(f"New quota {new_quota_str} got applied!!")
+
+    @bugzilla("2188032")
+    @pytest.mark.parametrize(
+        argnames="amount,interface,quota",
+        argvalues=[
+            pytest.param(
+                *[1, "RGW-OC", {"maxObjects": "10"}],
+                marks=[
+                    tier2,
+                ],
+            ),
+        ],
+    )
+    def test_obc_quota_full_alert(
+        self,
+        rgw_bucket_factory,
+        rgw_obj_session,
+        awscli_pod_session,
+        test_directory_setup,
+        threading_lock,
+        amount,
+        interface,
+        quota,
+    ):
+        """
+        This test will verify the prometheus alerts
+        when the OBC quota is reached
+
+        """
+
+        # create the bucket
+        bucket_name = rgw_bucket_factory(amount, interface, quota=quota)[0].name
+        logger.info(f"created rgw bucket {bucket_name} with quota {quota}")
+
+        # fill the bucket with about 90% of maxObjects
+        number_of_objects = int(quota["maxObjects"])
+        write_random_test_objects_to_bucket(
+            awscli_pod_session,
+            bucket_name,
+            test_directory_setup.origin_dir,
+            amount=(number_of_objects * 90) // 100,
+            mcg_obj=OBC(bucket_name),
+        )
+        logger.info(f"Filled bucket {bucket_name} with 90% maxObjects capacity")
+
+        # wait for obc full alert to occur and verify
+        prometheus = PrometheusAPI(threading_lock=threading_lock)
+        alerts = [
+            alert
+            for alert in prometheus.wait_for_alert(
+                name=constants.ALERT_OBC_QUOTA_OBJECTS_ALERT,
+                state="firing",
+                timeout=600,
+            )
+            if alert.get("labels").get("objectbucketclaim") == bucket_name
+        ]
+
+        if len(alerts) == 0:
+            assert False, (
+                f"Alert {constants.ALERT_OBC_QUOTA_OBJECTS_ALERT} doesn't seem to occur "
+                f"despite the bucket being 90% full"
+            )
+
+        alert_desc = (
+            f"ObjectBucketClaim {bucket_name} has crossed 80% "
+            f"of the size limit set by the quota(objects)"
+        )
+        for alert in alerts:
+            if alert_desc in alert.get("annotations").get("description"):
+                logger.info(
+                    f"Verified the alert {constants.ALERT_OBC_QUOTA_OBJECTS_ALERT}"
+                )
+                break
+            else:
+                assert (
+                    False
+                ), f"Alert {constants.ALERT_OBC_QUOTA_OBJECTS_ALERT} doesn't seem have expected format"
```

Comment:
This loop either breaks or fails the test on the first iteration, thus only checking the first alert on the list.

If you wish to check all the alerts, the following would fix the bug:
```suggestion
        for alert in alerts:
            assert (
                False
            ), f"Alert {constants.ALERT_OBC_QUOTA_OBJECTS_ALERT} doesn't seem have expected format"     
        logger.info(
            f"Verified the alert {constants.ALERT_OBC_QUOTA_OBJECTS_ALERT}"
            )
```

And If your intention is to sample only one of the alerts, you can use the following instead for simplicity:
```suggestion

        assert alert_desc in alerts[0].get("annotations").get("description"), (
            f"Alert {constants.ALERT_OBC_QUOTA_OBJECTS_ALERT} doesn't seem have expected format"
        )
        logger.info(
            f"Verified the alert {constants.ALERT_OBC_QUOTA_OBJECTS_ALERT}"
        )       
```

I

Reply:
@sagihirshfeld I want to sample all the alerts but your suggestion 1 doesnt work right? it straight way asserts. To give more details, I want to check all the alert in the alerts list which is filtered to have OBC Quota reached alerts only. As part of check, i just verify that it adheres the expected format. 

To achieve this, i think i need to replace break with continue and move the log confirming the successful verification out of the for loop.

Reply:
below should work:

>         for alert in alerts:
            assert alert_desc in alert.get("annotations").get("description"), f"Alert {constants.ALERT_OBC_QUOTA_OBJECTS_ALERT} doesn't seem have expected format"
        logger.info(
            f"Verified the alert {constants.ALERT_OBC_QUOTA_OBJECTS_ALERT}"
        )

Reply:
You're correct my first solution doesn't work - while editing and formatting I must've messed up and reused the original assertion by mistake. Please use the code in your last comment instead  👍